src/util/simplify_expr.cpp

```diff
@@ -1744,6 +1744,31 @@ simplify_exprt::simplify_byte_extract(const byte_extract_exprt &expr)
     }
   }
 
+  // byte_extract(byte_update(root, offset_u, value), offset_e) so that the
+  // extracted range fully lies within the update value simplifies to
+  // byte_extract(value, offset_e - offset_u)
+  if(
+    expr.op().id() == ID_byte_update_big_endian ||
+    expr.op().id() == ID_byte_update_little_endian)
+  {
+    const byte_update_exprt &bu = to_byte_update_expr(expr.op());
+    const auto update_offset = numeric_cast<mp_integer>(bu.offset());
+    const auto update_size = pointer_offset_bits(bu.value().type(), ns);
+
+    if(
+      el_size.has_value() && update_offset.has_value() &&
```

Comment:
Nit: This is a chunky condition. I would *either* suggest dressing it with a comment to make parsing easier, *or* just break the logical bits into named boolean values above the expression and combine them logically within it, for example something like:

```c++
const bool values_present = el_size.has_value() && update_offset.has_value() && update_size.has_value();
// ...

if (values_present && within_offsets && ..)
{
...
```

Reply:
Done, as a second refactoring commit.